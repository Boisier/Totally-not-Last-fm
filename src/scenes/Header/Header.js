// React
import React, { Component } from 'react'

// Scene dependencies
import 'style/scenes/Header.scss'

export default class extends Component {
  render () {
    return (
      <header>
        <h3>Totally not <span className = "red">Last fm</span></h3>
      </header>
    )
  }
}
