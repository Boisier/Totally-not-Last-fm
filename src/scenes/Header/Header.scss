@import '~style/root';

// Hamburger & hamburger color
$hamburger-layer-color: #ff0000;
$hamburger-layer-width: 30px;
$hamburger-layer-height: 3px;
@import '~hamburgers/_sass/hamburgers/hamburgers';

$header-transition: .15s;

header
{
  font-family:'Rubik', sans-serif;
  display: flex;
  flex-flow: row wrap;
  align-items: flex-start;
  overflow: hidden;

  position: fixed;
  z-index:1000;
  top: 0;
  left: 0;

  height: $header-height;
  width: 100%;
  padding: 0 5rem;

  background-color: $pink;

  transition: $header-transition;

  h3
  {
    display: block;
    width: 80%;
    height: $header-height;
    margin: 0;

    color: $grey;
    text-align: left;
    text-transform: uppercase;
    font-style: italic;
    font-size: 1.5rem;
    line-height: $header-height;

    transition: $header-transition;

    cursor: pointer;

    .red {
      padding-left: .35rem;
      letter-spacing: -0.1em;
      color: red;
    }

    .page-title {
      padding-left: .35rem;
      color: #000;
    }
  }

  h5
  {
    height: $header-height;
    margin: 0;

    .hamburger
    {
      display: inline-block;
      padding: 14px;

      outline: none;

      .hamburger-box
      {
        display: inline-block;
        vertical-align: middle;
      }

      .hamburger-label
      {
        display: inline-block;
        margin-left: .5rem;
        vertical-align: middle;

        color: $hamburger-layer-color;
        text-transform: uppercase;
      }
    }
  }


  &.menu-opened
  {
    height: 100%;

    h3
    {
      padding-top: 1rem;
      font-size: 3rem;
    }

    .page-title
    {
      opacity: 0;
    }
  }
}
