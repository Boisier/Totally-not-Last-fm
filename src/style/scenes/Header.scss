@import '../root';

header
{

  @import url('https://fonts.googleapis.com/css?family=Rubik:300,300i,400,400i,500,500i,700,700i,900,900i');

  font-family:'Rubik', sans-serif;
  display: block;
  position: fixed;
  top: 0;
  left: 0;

  height: $header-height;
  width: 100%;
  z-index:1000;
  background-color: $pink;

  h3
  {
    width:23%;
    color: $grey;
    margin: 10px;
    padding-left: 5em;
    text-align: left;
    font-style: italic;
    text-transform: uppercase;
    cursor: pointer;

    .red {
      letter-spacing: -0.1em;
      color: red;
    }
  }
}

// Update body with header height
body
{
  //padding-top: $header-height;
}
