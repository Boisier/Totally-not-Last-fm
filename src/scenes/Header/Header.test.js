import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Header from './Header'

it('renders LinkBlock correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  ).toJSON()
  expect(component).toMatchSnapshot()
})
