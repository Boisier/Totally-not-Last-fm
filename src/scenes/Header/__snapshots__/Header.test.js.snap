// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LinkBlock correctly 1`] = `
<header>
  <h1>
    Totally not Last fm online
  </h1>
</header>
`;
